dataRetriever/txpool/shardedTxPool.go

```diff
@@ -389,3 +391,44 @@ func (txPool *shardedTxPool) routeToCacheUnions(cacheID string) string {
 
 	return cacheID
 }
+
+func (txPool *shardedTxPool) getMempool() txCache {
+	shard, ok := txPool.backingMap[strconv.Itoa(int(txPool.selfShardID))]
+	if !ok {
+		return nil
+	}
+
+	return txPool.getTxCache(shard.CacheID)
+}
+
+func (txPool *shardedTxPool) MempoolCleanup(session interface{}, randomness uint64, maxNum int, maxTime time.Duration) bool{
```

Comment:
After discussing a bit with Adi - maybe we should stick to our previous term, `TxPool` - and not mention "mempool" at all (I know the term is referred in some existing code, as well)?

Here, maybe `CleanupSelfShardTxCache()`? Let's double check with Adi.